FlowCrypt/Functionality/Mail Provider/Backup Provider/Imap+Backup.swift

```diff
@@ -51,17 +51,20 @@ extension Imap: BackupProvider {
                 return msgs.map { msg in MsgContext(path: uidsContext.path, msg: msg) }
             }
 
+            // in case there are no messages return empty data
+            // user will be prompted to create new backup
             guard messageContexts.isNotEmpty else {
-                throw BackupError.missedMessages
+                return Data()
             }
 
             let attContext = messageContexts.flatMap { msgContext -> [AttContext] in
                 guard let parts = msgContext.msg.attachments() as? [MCOIMAPPart] else { assertionFailure(); return [] }
                 return parts.map { part in AttContext(path: msgContext.path, msg: msgContext.msg, part: part) }
             }
 
+            // in case there are no attachments return empty data
             guard attContext.isNotEmpty else {
-                throw BackupError.missedAttributes
+                return Data()
             }
 
             let dataArr = try attContext.map { attContext -> Data in
```

Comment:
I suppose on the line below `fetchMsgAttribute` should be called `fetchMsgAttachment`. Also above, for clarity, rename `attContext` to `attachmentContext`. I'll make a separate issue.

Reply:
fixed in scope of this PR